lib/iris/cube.py

```diff
@@ -789,7 +788,7 @@ class CubeAttrsDict(MutableMapping):
 
     def __init__(
         self,
-        combined: Optional[Union[Mapping, str]] = "__unspecified",
+        combined: Optional[Mapping] = None,
```

Comment:
This looks good to me, and certainly makes things simpler. @pp-mo am I missing anything?

Reply:
@pp-mo said there might be a good reason for this, but equally it could be a hangover from a previous implementation, especially since all the tests still pass when it is removed.

I'll give this a few days and if there are no further comments by then I will approve @bouweandela's suggestion.

Reply:
I'm looking into this.
Actually, I created some additional tests for untested behaviour, and it does look this change preserves all the behaviour, even the undocumented (!)
So, good spot ! 👍 

Having said which, I think there are some holes in the current scheme : I will propose something new to tighten that up.
But it shouldn't get in the way of this, so I will call this resolved.